test/smoke.test.ts

```diff
@@ -1,19 +1,111 @@
 import fetchMock from "fetch-mock";
-import { RequestOptions } from "https";
-import { HttpsProxyAgent } from "https-proxy-agent";
+import { createServer, type Server } from "https";
+import { Octokit, getProxyAgent, customFetch } from "../src";
+import { ProxyAgent } from "undici";
+
+// mock undici such that we can substitute our own fetch implementation
+// but use the actual ProxyAgent implementation for most tests. the
+// exception is "should call undiciFetch with the correct dispatcher"
+// where we want to validate that a mocked ProxyAgent is passed through
+// to undici.fetch.
+jest.mock("undici", () => {
+  return {
+    fetch: jest.fn(),
+    ProxyAgent: jest.requireActual("undici").ProxyAgent,
+  };
+});
+const undici = jest.requireMock("undici");
 
-import { Octokit } from "../src";
+describe("Smoke test", () => {
+  let server: Server;
 
-jest.mock("https-proxy-agent");
+  beforeAll((done) => {
+    server = createServer(
+      {
+        requestCert: false,
+        rejectUnauthorized: false,
+      },
+      (request: any, response: any) => {
+        expect(request.method).toEqual("GET");
+        expect(request.url).toEqual("/");
+
+        response.writeHead(200);
+        response.write("ok");
+        response.end();
+      },
+    );
+
+    server.listen(0, done);
+  });
 
-describe("Smoke test", () => {
   beforeEach(() => {
     delete process.env.GITHUB_TOKEN;
     delete process.env.INPUT_GITHUB_TOKEN;
     delete process.env.GITHUB_ACTION;
     delete process.env.GITHUB_API_URL;
     delete process.env.HTTPS_PROXY;
     delete process.env.https_proxy;
+    delete process.env.HTTP_PROXY;
+    delete process.env.http_proxy;
+  });
+
+  afterAll((done) => {
+    server.close(done);
+    jest.unmock("undici");
+  });
+
+  it("should return a ProxyAgent for the httpProxy environment variable", () => {
```

Comment:
These tests monitor the `getProxyAgent` functionality. Maybe they're unnecessary.

Reply:
I think these tests are not necessary, the integration test below should cover all that's needed.

Reply:
Removed!

Reply:
Oops, apparently code coverage was affected and lines 20 and 28 in index.ts were being tested by these tests. I've added them back accordingly.